mitodl/ocw-studio
# Add info to backend repo required for publish/preview pipelines

This is probably no longer necessary since the pipelines will be created from ocw-studio (#366)

#### Pre-Flight checklist

#### What are the relevant tickets?
Closes #319 

#### What's this PR do?
- Changes the initial commit for new repos to be a `site_config_vars.yml` file containing the following info, formatted to be easily parsed by a bash script in a concourse publishing pipeline:

    ```
    CONFIG_PATH={website.starter.path}
    CONFIG_SLUG={website.starter.slug}
    SITE_SLUG={website.name}"
    ```
- Adds a `backpopulate_site_configs` management command to add the above file to one website or all websites with existing repos.
- Adds a migration to the `WebsiteStarter` model so that `slug` and `path` must be unique together, not `slug` alone (different repos might have the same slug).

#### How should this be manually tested?
- Set up github integration by following the README
- Create a new `Website` via the UI.  
- Check the website's git repo and verify that there is a `site_config_vars.yml` file there with the correct content.
- Run the `backpopulate_site_configs` command and verify that all the `Website` objects for which repos already exist have a `site_config_varsyml` file afterward.

#### Any background context you want to provide?
The concourse pipeline that will be generated for each `Website` will be running the following:
```
hugo --config ../ocw-hugo-projects/((site-config))/config.yaml --themesDir ../ocw-hugo-themes/theme
```
and then uploading files to an S3 bucket with a path dependent on the `Website.name` value:
```
destination:           
   - dir: s3-remote:((ocw-bucket))/courses/((course-id))
```

This file is meant to provide the info needed to replace `((site-config))` and `((course-id))` values above.

This is assuming that all the websites we want to publish via concourse will be associated with a `WebsiteStarter` of `source=github` and that the repo will have 1 or more yaml files under its path/slug containing content that looks something like this:

```
---
baseUrl: "/"
enableRobotsTXT: true
languageCode: en-usrelative
URLs: false
title: MIT OpenCourseWare
theme: ["base-theme", "course"]
```

Pipelines should not be generated if such a file cannot be found at the specified github url/path (but that is a separate issue).

I originally thought about making this file a required `WebsiteContent` object and handling it that way, but ultimately felt like it didn't quite fit that use case.  The content of this file is info that currently cannot be modified after a  `Website` is created.
